internal/ackhandler/packet_number_generator.go

```diff
@@ -34,25 +33,45 @@ func (p *sequentialPacketNumberGenerator) Pop() protocol.PacketNumber {
 	return next
 }
 
+type rng struct {
+	buf [4]byte
+}
+
+func (r *rng) Int31() int32 {
+	rand.Read(r.buf[:])
+	return int32(binary.BigEndian.Uint32(r.buf[:]) & ^uint32(1<<31))
+}
+
+// copied from the standard library math/rand implementation of Int63n
+func (r *rng) Int31n(n int32) int32 {
+	if n&(n-1) == 0 { // n is power of two, can mask
+		return r.Int31() & (n - 1)
+	}
+	max := int32((1 << 31) - 1 - (1<<31)%uint32(n))
+	v := r.Int31()
+	for v > max {
```

Comment:
Maybe increase the fallback by powers of two? That would allow you to drop this loop / skip the loop in crypto/rand.Int()?

Reply:
Not sure I understand your comment. tbh, I just copied this from the `math/rand` implementation of `Int63n`. `max` should be very close to `math.MaxInt32`, so this loop is very unlikely to be executed.